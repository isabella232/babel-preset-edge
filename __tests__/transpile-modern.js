import buildPreset from "../src"

import { check, fixtures, titles } from "./core"

describe("Transpile: Modern", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    transpile: "modern"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
