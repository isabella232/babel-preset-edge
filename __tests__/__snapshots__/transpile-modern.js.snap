// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpile: Modern comments 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

function translate(value) {
  return _camelCase(value);
}"
`;

exports[`Transpile: Modern deadcode 1`] = `
"const STATIC = false;
before();
after();"
`;

exports[`Transpile: Modern emotion 1`] = `
"import { css as _css } from \\"emotion\\";

const className =
/*#__PURE__*/
_css({
  name: \\"1vdp66k-className\\",
  styles: \\"color:hotpink;label:className;\\"
});

function SomeComponent() {
  return React.createElement(\\"div\\", {
    className: className
  }, \\"Some hotpink text.\\");
}"
`;

exports[`Transpile: Modern es2015: classes 1`] = `
"export class Component extends Other {
  constructor() {
    super();
    this.flag = true;
  }

  main() {
    alert(this.foo);
  }

}"
`;

exports[`Transpile: Modern es2015: for_off 1`] = `
"import \\"core-js/modules/web.dom-collections.iterator\\";

function test() {
  for (const taskName of tasks) {
    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Transpile: Modern es2015: generator 1`] = `
"function* idMaker() {
  var index = 0;

  while (true) yield index++;
}"
`;

exports[`Transpile: Modern es2015: set 1`] = `
"import \\"core-js/modules/web.dom-collections.iterator\\";
new Set([1, 2, 3]);"
`;

exports[`Transpile: Modern es2016: exponentiation 1`] = `"var result = 2 ** 3;"`;

exports[`Transpile: Modern esnext: async 1`] = `
"import \\"core-js/modules/es.promise\\";

async function wait() {
  await loadData();
}"
`;

exports[`Transpile: Modern esnext: async_loop 1`] = `
"import \\"core-js/modules/es.promise\\";
import \\"core-js/modules/web.dom-collections.iterator\\";

async function executeTasks(tasks) {
  for (const taskName of tasks) {
    try {
      await executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Transpile: Modern esnext: catch_bind 1`] = `
"try {
  throw 0;
} catch (_unused) {
  doSomethingWhichDoesntCareAboutTheValueThrown();
} finally {
  doSomeCleanup();
}"
`;

exports[`Transpile: Modern esnext: class properties 1`] = `
"export class Component {
  constructor() {
    this.onClick = () => {};
  }

}"
`;

exports[`Transpile: Modern esnext: decorators 1`] = `
"var _class;

let MyClass = annotation(_class = class MyClass {}) || _class;

function annotation(target) {
  target.annotated = true;
}"
`;

exports[`Transpile: Modern esnext: dynamic import 1`] = `
"import \\"core-js/modules/es.promise\\";
import(
/*webpackChunkName:'lodash-wBYk8'*/
\\"lodash\\").then(lodash => {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Transpile: Modern esnext: object spread 1`] = `
"let original = {
  bar: 10
},
    variant = Object.assign({
  foo: 1
}, original);"
`;

exports[`Transpile: Modern loadable 1`] = `
"import \\"core-js/modules/es.promise\\";
import loadable from 'loadable-components';
export const Home = loadable({
  chunkName() {
    return \\"Home-JfPYC\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Home-JfPYC\\" */
  \\"./views/Home\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return require('path').resolve(__dirname, \\"./views/Home\\");
  }

});
export const About = loadable({
  chunkName() {
    return \\"About-8yDxR\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"About-8yDxR\\" */
  \\"./views/About\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return require('path').resolve(__dirname, \\"./views/About\\");
  }

});
export const Contact = loadable({
  chunkName() {
    return \\"Contact-AIf6m\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Contact-AIf6m\\" */
  \\"./views/Contact\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return require('path').resolve(__dirname, \\"./views/Contact\\");
  }

});"
`;

exports[`Transpile: Modern loadable magic-comments 1`] = `
"import \\"core-js/modules/es.promise\\";
import loadable from 'loadable-components';
export const Home = loadable({
  chunkName() {
    return \\"Home-JfPYC\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"Home-JfPYC\\" */
  \\"./views/Home\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return require('path').resolve(__dirname, \\"./views/Home\\");
  }

});
export const About = loadable({
  chunkName() {
    return \\"About\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"About\\" */
  \\"./views/About\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return require('path').resolve(__dirname, \\"./views/About\\");
  }

});
export const Contact = loadable({
  chunkName() {
    return \\"Contact-AIf6m\\";
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackMode: \\"lazy\\", webpackFoo: \\"bar\\", webpackChunkName: \\"Contact-AIf6m\\" */
  \\"./views/Contact\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return require('path').resolve(__dirname, \\"./views/Contact\\");
  }

});"
`;

exports[`Transpile: Modern loadable wildcard 1`] = `
"import \\"core-js/modules/es.promise\\";
import \\"core-js/modules/es.string.replace\\";
import loadable from 'loadable-components';
const page = \\"About\\",
      lang = \\"de_DE\\";
export const views = loadable({
  chunkName() {
    return (\\"views-\\" + page).replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"views-[request]\\" */
  \\"./views/\\" + page),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/\\" + page);
    }

    return require('path').resolve(__dirname, \\"./views/\\" + page);
  }

});
export const leanIntl = loadable({
  chunkName() {
    return (\\"lean-intl-\\" + lang + \\"-data\\").replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"lean-intl-[request]-data\\" */
  \\"./lean-intl/\\" + lang + \\"/data\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./lean-intl/\\" + lang + \\"/data\\");
    }

    return require('path').resolve(__dirname, \\"./lean-intl/\\" + lang + \\"/data\\");
  }

});
export const reactIntl = loadable({
  chunkName() {
    return (\\"react-intl-\\" + lang + \\"-data\\").replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },

  isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },

  requireAsync: () => import(
  /* webpackChunkName: \\"react-intl-[request]-data\\" */
  \\"./react-intl/\\" + lang + \\"/data\\"),

  requireSync(props) {
    const id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },

  resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./react-intl/\\" + lang + \\"/data\\");
    }

    return require('path').resolve(__dirname, \\"./react-intl/\\" + lang + \\"/data\\");
  }

});"
`;

exports[`Transpile: Modern lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`Transpile: Modern lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`Transpile: Modern lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`Transpile: Modern macro: gql 1`] = `
"const query = {
  kind: \\"Document\\",
  definitions: [{
    kind: \\"OperationDefinition\\",
    operation: \\"query\\",
    name: {
      kind: \\"Name\\",
      value: \\"User\\"
    },
    variableDefinitions: [],
    directives: [],
    selectionSet: {
      kind: \\"SelectionSet\\",
      selections: [{
        kind: \\"Field\\",
        name: {
          kind: \\"Name\\",
          value: \\"user\\"
        },
        arguments: [{
          kind: \\"Argument\\",
          name: {
            kind: \\"Name\\",
            value: \\"id\\"
          },
          value: {
            kind: \\"IntValue\\",
            value: \\"5\\"
          }
        }],
        directives: [],
        selectionSet: {
          kind: \\"SelectionSet\\",
          selections: [{
            kind: \\"Field\\",
            name: {
              kind: \\"Name\\",
              value: \\"lastName\\"
            },
            arguments: [],
            directives: []
          }, {
            kind: \\"FragmentSpread\\",
            name: {
              kind: \\"Name\\",
              value: \\"UserEntry1\\"
            },
            directives: []
          }]
        }
      }]
    }
  }],
  loc: {
    start: 0,
    end: 79,
    source: {
      body: \\"\\\\n  query User {\\\\n    user(id: 5) {\\\\n      lastName\\\\n      ...UserEntry1\\\\n    }\\\\n  }\\\\n\\",
      name: \\"GraphQL request\\",
      locationOffset: {
        line: 1,
        column: 1
      }
    }
  }
};"
`;

exports[`Transpile: Modern macro: preval 1`] = `"const one = 1;"`;

exports[`Transpile: Modern promise 1`] = `
"import \\"core-js/modules/es.promise\\";
export default function () {
  return new Promise(resolve => {
    return resolve(42);
  });
}"
`;

exports[`Transpile: Modern react: jsx 1`] = `
"function render() {
  return React.createElement(\\"h1\\", null, \\"Hello\\");
}"
`;

exports[`Transpile: Modern react: jsx spread 1`] = `
"function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props), \\"Hello\\");
}"
`;

exports[`Transpile: Modern react: proptypes 1`] = `
"import React from \\"react\\";
import PropTypes from \\"prop-types\\";

class MyComponent extends React.Component {
  render() {
    return React.createElement(\\"div\\", null, this.props.userName);
  }

}

MyComponent.propTypes = {
  userName: PropTypes.string
};"
`;

exports[`Transpile: Modern require context 1`] = `
"if (process.env.NODE_ENV === \\"test\\") {
  require('babel-plugin-require-context-hook/register')();
}

const loader = require.context(__dirname, false, /\\\\.js/);"
`;
